carsons/carsons.py

```diff
@@ -65,6 +65,24 @@ def perform_kron_reduction(z_primitive: ndarray, dimension=3) -> ndarray:
     return Z_abc
 
 
+def calculate_self_impedance(Z):
+    return (Z[0, 0] + Z[1, 1] + Z[2, 2])/3
+
+
+def calculate_mutual_impedance(Z):
+    return (Z[0, 1] + Z[1, 2] + Z[0, 2])/3
+
+
+def calculate_sequence_impedances(Z):
```

Comment:
it seems to me there are two different calculations being lumped together here -- in Kersting 4.3 (i'm using the 4th edition), the text uses the formula `z₀₁₂ = Aₛ⁻¹ ⋅ z_abc ⋅ Aₛ` to compute a 3x3 matrix of sequence impedances -- it doesn't simplify to a z1/z0 pair. and here, you are averaging to get a balanced impedance *before* converting to sequence impedance. so, it doesn't exactly match up, and it also precludes the use of `calculate_sequence_impedances`  to calculate unbalanced sequence impedances, which to me would be consistent with the phase-impedance function that returns a 3-phase matrix of impedance values. do you know if there is a reduction we could do on the 3-phase sequence impedance matrix that would allow us to support both 3-phase and single-phase?

Reply:
@AnjoMan, not sure, but I think taking the diagonal of Z012 will do what you want. At least it's equivalent for the case of balanced Zabc = [[z_s, z_m, z_m], [z_m, z_s, z_m], [z_s, z_s, z_m]]

Reply:
Got around to checking. The diagonals of Z012 are exactly equal to zs + 2*zm and zs - zm calculated over the averaged diagonal and off-diagonal elements for Zabc, even for asymmetric line configurations. Good to know!